zenodo/modules/spam/ext.py

```diff
@@ -0,0 +1,70 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of Zenodo.
+# Copyright (C) 2020 CERN.
+#
+# Zenodo is free software; you can redistribute it
+# and/or modify it under the terms of the GNU General Public License as
+# published by the Free Software Foundation; either version 2 of the
+# License, or (at your option) any later version.
+#
+# Zenodo is distributed in the hope that it will be
+# useful, but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Zenodo; if not, write to the
+# Free Software Foundation, Inc., 59 Temple Place, Suite 330, Boston,
+# MA 02111-1307, USA.
+#
+# In applying this license, CERN does not
+# waive the privileges and immunities granted to it by virtue of its status
+# as an Intergovernmental Organization or submit itself to any jurisdiction.
+
+"""Support and contact module for Zenodo."""
+
+from __future__ import absolute_import, print_function
+
+from werkzeug.utils import cached_property
+from zenodo.modules.spam.tasks import warm_up_cache
+from flask import current_app
+
+import joblib
+from . import config
+
+
+class ZenodoSpam(object):
+    """Zenodo support form."""
+
+    @cached_property
+    def model(self):
+        """Spam detection model."""
+        self._is_cache_loading = True
+        if not current_app.config.get('ZENODO_SPAM_MODEL_LOCATION'):
+            return None
+        return joblib.load(current_app.config['ZENODO_SPAM_MODEL_LOCATION'])
+
+    @property
+    def is_cache_loading(self):
+        """Flag in case another thread is loading already the model."""
+        return getattr(self, '_is_cache_loading', False)
+
+    def __init__(self, app=None):
+        """Extension initialization."""
+        if app:
+            self.init_app(app)
+
+    def init_app(self, app):
+        """Flask application initialization."""
+        self.app = app
+        self.init_config(app)
+        warm_up_cache.delay()
```

Comment:
This is not safe since we don't know if Flask-CeleryExt has been already initialized (i.e. maybe the RabbitMQ connection URL config is not yet loaded/set). An interesting alternative might be [Celery worker lifecycle signals](https://docs.celeryproject.org/en/v4.1.0/userguide/signals.html) (e.g.`worker_init`).

Reply:
Indeed that looks much cleaner. Maybe this way we can also have a guarantee that the code will run on all processes.